media_kit_video/lib/media_kit_video_controls/src/controls/material.dart

```diff
@@ -461,7 +488,7 @@ class _MaterialVideoControls extends StatefulWidget {
 class _MaterialVideoControlsState extends State<_MaterialVideoControls> {
   late bool mount = _theme(context).visibleOnMount;
   late bool visible = _theme(context).visibleOnMount;
-
+  late double _widgetWidth;
```

Comment:
This is a bad approach. Other 2 better ways

1. Just do it all in widget tree. Put in a stack above the video or whatever, three different gesture detectors in a row with different callbacks

2. Stay with one gesture detector. But get the rect on-tap using
```dart
(context.findRenderObject() as RenderBox).paintBounds
```

Reply:
i prefer the 2 approach and implemented it